### PR TITLE
Dropdown assets

### PR DESCRIPTION
**Pull request by ayoblvck**
fixes  "When a new hazard is added, the hazard is tagged against all stages . I.e demolition, construction, commission etc. Add in functionality which allows you to select i.e. construction, commission etc  from a pick list field

Instead of all categories selected by default, let user select the stage they want rather than unselecting them"

### CODE REVIEW

File: SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js [hunk 1]
```diff
@@ -729,10 +729,30 @@ tposdata = {
 
         hListItem.update();
         ctx().executeQueryAsync();
-        if (callback) {
+        if (callback && callback === Function) {
```
Comment (ayoblvck) [inline SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js:732]: This throws an error when call back is not a function hence the second condition

File: SiteAssets/pages/3.0/html/new.hazard.form.html [hunk 2]
```diff
@@ -328,37 +328,47 @@
...
+            console.log(data)
+            for (var i = 0; i < data.d.results.length; i++) {
+                if (data.d.results[i].Title != "Temporary Works Construction"){
+                sTerm += '<option class="qresult" data-entity="permanent" data-stage="' + data.d.results[i].ID + '" data-name="' + vn +
+                    '" data-value="' + v + '"><span>' + vn + ' - <span class="stge-option">' + data.d.results[i].Title + '</span>' +
+                    ' </span></option>';
+            }}
             nt = t + sTerm;
-            lbl = '<div class="tpos-lbl">Selected asset(s)</div>';
+            lbl = '<div class=""tpos-lbl"">Selected asset(s)</div>';
```
Comment (JHolland98) [inline SiteAssets/pages/3.0/html/new.hazard.form.html:351]: Remove one set of the double quotes here

File: SiteAssets/pages/3.0/html/new.hazard.form.html [hunk 3]
```diff
@@ -328,37 +328,47 @@
...
+            )
+
+                // Added this function to create a dropdown
+                $("#selectedstructures").multiselect({
+                    columns: 1,
+                    placeholder: 'select assets :',
+                    search: false,
+                    selectAll: true
+                })  
+            }}
```
Comment (JHolland98) [inline SiteAssets/pages/3.0/html/new.hazard.form.html:367]: Can we tab back lines 360 to 370 so they're aligned properly

File: SiteAssets/pages/3.0/html/new.hazard.form.html [hunk 4]
```diff
@@ -581,29 +592,43 @@
...
         const cdmHazardCoordinates = urlParams.get("cdmHazardCoordinates");
-
+        const selected_ele = document.getElementsByClassName("selected");
         var qr = $('.qresult');
+        var selection = [] //To store all selections
+        
+
+        for (i=0; i<selected_ele.length; i++){
+                    selection.push(selected_ele[i].innerText)
+                }
```
Comment (JHolland98) [inline SiteAssets/pages/3.0/html/new.hazard.form.html:602]: tab back lines 601 and 602 until they're lined correctly

File: SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js [hunk 5]
```diff
@@ -729,10 +729,30 @@ tposdata = {
...
             var fn = window[callback];
             fn(i);
         }
+        function onSuccess(){
+            if(callback === "addHazards"){
+                console.log("Success")
+                //let message appear after the last hazard is added
+                if(check[0] === (check[1] - 1)){
+                toastr.success(check[1] + " hazard(s) successfully added")
```
Comment (JHolland98) [inline SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js:741]: Remove the stuff about the check - it's better to have a toast for each successful hazard creation

File: SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js [hunk 6]
```diff
@@ -703,7 +703,7 @@ tposdata = {
                 });
             });
     },
-    set: function(lst, data, callback) {
+    set: function(lst, data, callback, check) {
```
Comment (JHolland98) [inline SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js:706]: remove the check parameter

File: SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js [hunk 7]
```diff
@@ -729,10 +729,30 @@ tposdata = {
...
+                }
+            }
+        }
+
+        function onFailure(sender, args){
+            console.log("failure");
+            if(callback === "addHazards"){
+                //let message appear after the last hazard is added
+                if(check[0] === (check[1] - 1)){
+                toastr.error("hazard(s) not added")
```
Comment (JHolland98) [inline SiteAssets/pages/3.0/scripts/CDM.DATA.CONTROLLER.js:751]: Remove the stuff about the check

**Review by JHolland98 — CHANGES_REQUESTED** [review]

**Review by JHolland98 — APPROVED** [review]

**Pull request merged**